### PR TITLE
fix: Fetch only top contributors

### PR DESCRIPTION
**Pull request by DanielPalafox**

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[Fetch only top contributors from the database](https://trello.com/c/DbRQ6XeT/8183-fetch-only-top-contributors-from-the-database)

**Review by amontenegro — APPROVED** [review]

File: orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java [hunk 1]
```diff
@@ -368,18 +368,23 @@ public WorkEntity merge(WorkEntity entity) {
...
 
     @Override
-    public List<Object[]> getWorksByOrcid(String orcid) {
-        //TODO @DanielPalafox fetch only the top_contributors_json after CLI finishes
+    public List<Object[]> getWorksByOrcid(String orcid, boolean topContributorsTogglz) {
+        String contributorsSQL = null;
+        if (topContributorsTogglz) {
+            contributorsSQL = " w.top_contributors_json ";
+        } else {
+            contributorsSQL = " (SELECT to_json(array_agg(row_to_json(t))) FROM (SELECT json_array_elements(json_extract_path(contributors_json, 'contributor')) AS contributor FROM work WHERE work_id=w.work_id limit 100) t) as contributors, ";
```
Comment (amontenegro) [inline orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java:376]: @DanielPalafox the comma at the end of this line will break the query right? could you please remove it?

Thanks

File: orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java [hunk 2]
```diff
@@ -400,7 +405,6 @@ public List<Object[]> getWorksByOrcid(String orcid) {
                 .addScalar("assertion_origin_client_source_id", StringType.INSTANCE)
                 .addScalar("date_created", TimestampType.INSTANCE)
                 .addScalar("last_modified", TimestampType.INSTANCE)
-                .addScalar("contributors", StringType.INSTANCE)
```
Comment (amontenegro) [inline orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java:403]: @DanielPalafox are you sure we dont need the scalar anymore?

**Review by amontenegro — CHANGES_REQUESTED** [review]

File: orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java (hunk 2 above)
Comment (DanielPalafox) [reply orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java:403]: Yes we need it you are rigth!. I already fix it!

File: orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java (hunk 1 above)
Comment (DanielPalafox) [reply orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java]: Ok, I will remove the unused comma

**Review by amontenegro — APPROVED** [review]

**Pull request merged**